apps/prairielearn/src/pages/instructorAssessmentInstance/instructorAssessmentInstance.html.ts

```diff
@@ -96,6 +97,50 @@ export function InstructorAssessmentInstance({
           navPage: '',
         })}
         <main id="content" class="container-fluid">
+          <div
+            class="modal fade"
```

Comment:
When we were using EJS we needed to put the modal directly into the HTML. With html templating, we can define a function for this and then include it at the correct point. In fact, we can do this for all the components of a page. Here's an example page that does this: https://github.com/PrairieLearnInc/PrairieTest/blob/main/src/pages/adminInstitutions/adminInstitutions.html.ts

Observe that the `DeleteInstitutionModal` is defined in [a function](https://github.com/PrairieLearnInc/PrairieTest/blob/a5056d7b239dd8bb2eed964e23fbad7e5f578b85/src/pages/adminInstitutions/adminInstitutions.html.ts#L293) and then included [into the main page](https://github.com/PrairieLearnInc/PrairieTest/blob/a5056d7b239dd8bb2eed964e23fbad7e5f578b85/src/pages/adminInstitutions/adminInstitutions.html.ts#L54). When copying this pattern, however, don't copy the use of a separate structure for the `DeleteInstitutionModalProps` -- just put them inline in the function as usual.

The above page also defines a [separate function](https://github.com/PrairieLearnInc/PrairieTest/blob/a5056d7b239dd8bb2eed964e23fbad7e5f578b85/src/pages/adminInstitutions/adminInstitutions.html.ts#L64) for the main content card and then includes it [in the page](https://github.com/PrairieLearnInc/PrairieTest/blob/a5056d7b239dd8bb2eed964e23fbad7e5f578b85/src/pages/adminInstitutions/adminInstitutions.html.ts#L55). I like this pattern because it keeps the page structure easy to understand and isolates the main content blocks (cards, modals, etc) into separate functions. I see that you've already done this for the `EditTotalScorePercForm` in the current page, so you can use the same pattern for the modal and cards.

Reply:
I have now updated this to use a a modal component in [cd0e458](https://github.com/PrairieLearn/PrairieLearn/pull/9578/commits/cd0e45818018ecb9941cf60705e037044a94db42). Thanks for the detailed explanation!